packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -762,9 +762,16 @@ let dynamicImportedComponentCounter = 0;
 /**
  * create a dynamic component via `<x-foo lwc:dynamic={Ctor}>`
  */
-function dc(
+type LightningElementConfig = {
+    constructor: LightningElementConstructor;
+    props: Record<string, any>;
+};
+function isLightningElementConfig(obj: any): obj is LightningElementConfig {
+    return !!obj.props;
+}
+export function dc(
     sel: string,
-    Ctor: LightningElementConstructor | null | undefined,
+    Ctor: LightningElementConstructor | LightningElementConfig | null | undefined,
```

Comment:
how many people are using this feature today in platform? can we migrate them?

Reply:
the ask is really whether or not we have the level of control in place to avoid having to support two APIs.